### DISCUSSION TITLE
Unable to Save Trivy Report to Host Directory

### DISCUSSION THREAD

**Discussion by gulammoideen** [aquasecurity/trivy#8004] [Q&A]
### Question

I am integrating Trivy with Jenkins using the Docker version. Each pipeline job runs on a Kubernetes-based Jenkins agent.

Since Trivy in Docker stores reports inside the container, I’ve mounted the host path (/home/jenkins/agent) to the container path (/output) to retrieve the report after the scan.
**Trivy Command Used:**
```
sudo docker run --rm \
  -v /var/run/docker.sock:/var/run/docker.sock \
  -v /home/jenkins/agent:/output \
  aquasec/trivy:0.57.1 image --format template \
  --template @contrib/html.tpl \
  -o /output/trivy-report-test.html \
  --db-repository public.ecr.aws/aquasecurity/trivy-db:2 <image-name>
```
**Problem:**
Despite running the above command, the report file is not saved in the host path. Here's the error message from the pipeline log:
```
+ ls -l /home/jenkins/agent/trivy-report-test.html
ls: cannot access '/home/jenkins/agent/trivy-report-test.html': No such file or directory
```
**Pipeline Log Snippet:**
```
17:38:28  2024-11-27T12:08:27Z  INFO  [vulndb] Need to update DB
17:38:28  2024-11-27T12:08:27Z  INFO  [vulndb] Downloading vulnerability DB...
...
17:38:38  2024-11-27T12:08:37Z  INFO  [python-pkg] Detecting vulnerabilities...
17:38:38  + ls -l /home/jenkins/agent/trivy-report-test.html
17:38:38  ls: cannot access '/home/jenkins/agent/trivy-report-test.html': No such file or directory
```
**Expected Behavior:**
The Trivy report should be saved in /home/jenkins/agent/trivy-report-test.html on the host machine.

**Steps to Reproduce:**
Run the Trivy command (as provided above) within a Jenkins pipeline.
Check the host directory for the report.

**Additional Context:**
Jenkins agents run as Kubernetes pods.
Permissions on /home/jenkins/agent allow read and write operations.

Please let me know if you need more details or any specific configurations.

### Target

Container Image

### Scanner

Vulnerability

### Output Format

Template

### Mode

Standalone

### Operating System

Jenkins agent (linux)

### Version

```bash
aquasec/trivy:0.57.1
```

**Comment by itaysk** [comment]
closing as this doesn't seem to be and issue relating to Trivy, rather a Docker/Jenkins usage
